templates/web/pages/product_edit/product_edit_form_packagings.tt.html

```diff
@@ -0,0 +1,182 @@
+        <div class="large-1 columns">[% lang("packaging_quantity_per_unit") %]</div>
+    </div>
+
+[% FOREACH packaging IN product.packagings %]
+    [% SET packaging_max = loop.index() %]
+    [% SET packaging_id = "packaging_" _ loop.index() %]
+    <div class="row packagings_components_edit_row" id="[% packaging_id %]_row">
+        <div class="small-12 large-1 columns">
+
+            <label class="hide-for-large-up" for="[% packaging_id %]_number_of_units">[% lang("packaging_number_of_units") %]</label>
+
+            <div class="row">
+                <div class="small-12 large-4 columns delete_packaging_column">
+                    <a class="delete_packaging" id="[% packaging_id %]_delete">
+                        <span class="material-icons">
+                            delete
+                        </span>
+                    </a>
+                </div>
+
+                <div class="small-12 large-8 columns">
+                    <input id="[% packaging_id %]_number_of_units" name="[% packaging_id %]_number_of_units" type="text" value="[% packaging.number_of_units %]">
+                </div>
+            </div>
+        </div>
+
+        [% FOREACH property IN ["shape", "material", "recycling"] %]
+
+            [% IF property == "shape" %]
+                [% SET taxonomy = "packaging_shapes" %]
+            [% ELSIF property == "material" %]
+                [% SET taxonomy = "packaging_materials" %]
+            [% ELSIF property == "recycling" %]
+                [% SET taxonomy = "packaging_recycling" %]
+            [% END %]
+
+            [% SET select2_id = packaging_id _ "_$property" %]
+
+        <div class="small-12 large-3 columns">
+            <label class="hide-for-large-up" for="[% select2_id %]">[% lang("packaging_$property") %]</label>
+            <select id="[% select2_id %]" name="[% select2_id %]" class="form-control [% taxonomy %]_select2">
+                <option></option>
+            </select>
+        </div>
+
+    [% IF packaging_max > 0 %]
+<initjs>
+\$("#[% select2_id %]").select2({
+    placeholder: "[% lang("packaging_$property") %]",
+    data: [% taxonomy %]_select2_options,
+    allowClear:true,
+    tags: true
+});
+
+[% IF packaging.$property.defined %]
+// Set the value, creating a new option if necessary
+var value = "[% display_taxonomy_tag(taxonomy, packaging.$property) %]";
+if (\$("#[% select2_id %]").find("option[value='" + value.replace(/'/g, "\\'") + "']").length) {
+    \$("#[% select2_id %]").val(value).trigger('change');
+} else { 
+    // Create a DOM Option and pre-select by default
+    var newOption = new Option(value, value, true, true);
+    // Append it to the select
+    \$("#[% select2_id %]").append(newOption).trigger('change');
+} 
+[% END %]
+
+\$("#[% packaging_id %]_delete").click(function(event) {
+    event.stopPropagation();
+    event.preventDefault();
+    \$("#[% packaging_id %]_row").remove();
+});
```

Comment:
we could also have this function only once by using a selector on a class, and getting the id of event target.